tests/ui/resolve/issue-56593-2.rs

```diff
@@ -0,0 +1,26 @@
+// check-pass
+
+use thing::*;
+
+#[derive(Debug)]
+pub enum Thing {
+    Foo,
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_thing() {
+        let thing = Thing::Foo;
```

Comment:
This is a breaking change too: `root::Thing` currently refers to `root::thing::Thing`, but it will refer to `root::thing::Thing` after this PR.

Reply:
The comment has a typo, but I assume you mean `root::thing::Thing` -> `root::Thing`.
It's also clearly a bugfix.

Crater will finish soon and show what actually breaks after these changes.